example/integration_test/util/assertion_helper.dart

```diff
@@ -3,11 +3,23 @@ import 'package:flutter/cupertino.dart';
 import 'package:flutter_test/flutter_test.dart';
 import '../extensions/string_extension.dart';
 
-// Assert expected text in a Text widget whose key starts with "nativeResponse_".
+/// Assert text in a Text widget whose key starts with "nativeResponse_" matches the expected value being passed.
 Future<void> assertNativeMessage(String suffix, String expected) async {
+  final actual = extractTextFromWidget(suffix);
+  assert(actual == expected, "Actual: $actual\nExpected: $expected");
+}
+
+/// Assert text in a Text widget whose key starts with "nativeResponse_" starts with the expected value being passed.
+Future<void> assertNativeMessageStartsWith(String suffix, String expected) async {
+  final actual = extractTextFromWidget(suffix);
+  assert(actual.startsWith(expected), "Actual: $actual\nExpected: $expected");
+}
+
+/// Return text from Text widget whose key starts with "nativeResponse_" and ends with suffix.
+String extractTextFromWidget(String suffix) {
```

Comment:
extractTextFromWidget should be private or moved to a dedicated class?

Reply:
I ended up making the method private